### PR TITLE
fix: fix invite flow while inviting via admin settings

### PR DESCRIPTION
**Pull request by shubham7saxena7**
> Pull Request Template
>
> Use this template to quickly create a well written pull request. Delete all quotes before creating the pull request.
>
## Description
the logic for handling user invitation on admin settings page was causing double emails on admin settings page. turns out, we cannot use isEnabled flag because that flag can also be set by other serviecs for an existing user.

#### PR fixes following issue(s)
Fixes # (issue number)
no issue - it is for a fix for existing buggy code in release
>
>

- Bug fix (non-breaking change which fixes an issue)
>
>
- [x] Manual
- [x] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java [hunk 1]
```diff
@@ -591,20 +593,24 @@ private Mono<Boolean> updateAdminUserPolicies(String oldAdminEmailsCsv, String o
...
-                        newUser.setIsEnabled(false);
-                        return newUser;
-                    }
-                    return existingUser;
-                }))
+        Flux<Tuple2<User, Boolean>> usersFlux = Flux.fromIterable(newUsers)
+                .flatMap(email -> userService
+                        .findByEmail(email)
+                        .flatMap(user -> {
+                            log.debug("User with email {} already exists", email);
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java:600]: Can the debug logs be removed before merging to release?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java [hunk 2]
```diff
@@ -591,20 +593,24 @@ private Mono<Boolean> updateAdminUserPolicies(String oldAdminEmailsCsv, String o
...
+                            return Mono.just(Tuples.of(user, false));
+                        })
+                        .switchIfEmpty(Mono.defer(() -> {
+                            log.debug("User with email {} not found", email);
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java:604]: Same as above. Please remove debug logs.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java [hunk 3]
```diff
@@ -591,20 +593,24 @@ private Mono<Boolean> updateAdminUserPolicies(String oldAdminEmailsCsv, String o
                 .collectList()
                 .flatMap(userUtils::removeSuperUser);
 
-        Flux<User> usersFlux = Flux.fromIterable(newUsers)
-                .flatMap(email -> userService.findByEmail(email).map(existingUser -> {
-                    if (existingUser == null) {
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java:596]: This is incorrect handling. findByEmail would not return anything.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java (hunk 1 above)
Comment (shubham7saxena7) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java]: done

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java (hunk 2 above)
Comment (shubham7saxena7) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java]: done

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java (hunk 3 above)
Comment (shubham7saxena7) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java:596]: yes, that was incorrect too. hence now used switchIfEmpty

**Pull request merged**